targets/TARGET_WIZNET/TARGET_W7500x/analogin_api.c

```diff
@@ -57,12 +55,8 @@ void analogin_init(analogin_t *obj, PinName pin)
     // Save pin number for the read function
     obj->pin = pin;
 
-    // The ADC initialization is done once
-    if (adc_inited == 0) {
-        adc_inited = 1;
-
-        ADC_Init();
-    }
+    // The ADC initialization
+    ADC_Init();
```

Comment:
I talked with WIZnet Justin and he said this

- Previously it worked fine. They thought ADC_Init() can be done only once and that was how it was implemented.
- During the CI ADC test, they found that previous ADC run had some 'lingering effect'.
- Now they realized that ADC should be hardware reset, which is done by calling ADC_Init() all the time when actual init is happening.
- This code is now better and more intuitive and correctly resets the ADC hardware.